dist/autocomplete.js

```diff
@@ -186,70 +205,73 @@
         self.model = undefined;
       }
     }
-
     /**
      * If clearOnNoSelection is true and no item has been
      * selected, clear any search text when user blurs the
      * input
      */
+
+
     function onFocusOut() {
       if (self.clearOnNoSelection && !self.model) {
         clearInput();
       }
+
       resetDropDownPostion();
     }
-
     /**
      * Check for drop down position when the element gets focus
      */
+
+
     function onFocusIn() {
       resetDropDownPostion();
 
       if (self.positionTimer) {
         $timeout.cancel(self.positionTimer);
-      }
-      // Allow plenty of time for the backend to respond to a request before firing
+      } // Allow plenty of time for the backend to respond to a request before firing
+
+
       self.positionTimer = $timeout(setDropDownPosition, self.positionTimeout * 10);
     }
-
     /**
      * @function resetDropDownPosition
      * Reset the drop down position of the box to under the input field
      */
+
+
     function resetDropDownPostion() {
       var dropDown = $element[0].querySelector('.angucomplete-dropdown');
-
       dropDown.style.top = null;
       dropDown.style.height = null;
     }
-
     /**
      * @function setDropDownPosition
      * Set the location of the dropdown. If there is not enough space at the bottom of the page the height will be set
      * above the input field. If the dropdown can not all be displayed in the top it is moved either above or below the
      * field, and resized to fit in the remaining space depending on where there in the most space
      */
+
+
     function setDropDownPosition() {
       var docTop = $window.pageYOffset + 125,
           docBottom = docTop + $window.innerHeight - 125 - 50,
           fieldTop = $element[0].getBoundingClientRect().top,
           fieldBottom = $element[0].getBoundingClientRect().bottom,
           dropDown = $element[0].querySelector('.angucomplete-dropdown'),
           dropDownTop = dropDown.getBoundingClientRect().height - 6;
-
       resetDropDownPostion();
 
       if (self.positionTimer) {
         $timeout.cancel(self.positionTimer);
         self.positionTimer = null;
-      }
+      } // Check that there is enough space at the bottom of the page
+
 
-      // Check that there is enough space at the bottom of the page
       if (docBottom < fieldBottom + dropDownTop) {
         // If no space at bottom move the dropdown list to the top
-        dropDown.style.top = (-dropDownTop).toString() + 'px';
+        dropDown.style.top = (-dropDownTop).toString() + 'px'; // Prevent top disappearing into menu area
 
```

Comment:
Again this comment has no relation to the line it has been added to. Surely it should address the following block ?